armory/scenarios/poison.py

```diff
@@ -302,7 +305,9 @@ def load_metrics(self):
                 "categorical_accuracy"
             )
 
-        self.benign_accuracy_per_class = {y:[] for y in np.unique(self.y_clean)} # store accuracy results for each class
+        self.benign_test_accuracy_per_class = {
```

Comment:
So there are two datasets in play here: a `train` and a `test`. The variable `self.y_clean` is really `self.y_clean_train`, so it seems like a potential issue to create the `self.benign_test_accuracy_per_class` keys from the `train` set. In the example I'm running, the `train` and `test` have exactly the same classes, but I imagine if they didn't (e..g if the test set only contains a subset of the train classes) then issues could arise. There's a similar part later on in `finalize_results()` that uses the `test` set keys to update both `test` and `train` statistics. It seems like there should be some further bifurcation of `test` and `train` variables to cover any case where the sets of labels aren't identical

Reply:
Ok makes sense, new commit allows for that case